light-zk.js/src/utxo.ts

```diff
@@ -630,27 +635,54 @@ export class Utxo {
         "aes-256-cbc",
         true,
       );
+
+      prefix = this.account.generateUtxoPrefixHash(commitment, PREFIX_LENGTH);
+
       if (!compressed) return ciphertext;
       const padding = sha3_256
         .create()
         .update(Uint8Array.from([...nonce, ...bytes_message]))
         .digest();
 
       // adding the 8 bytes as padding at the end to make the ciphertext the same length as nacl box ciphertexts of 120 bytes
-      return Uint8Array.from([...ciphertext, ...padding.subarray(0, 8)]);
+      return Uint8Array.from([
+        ...prefix,
+        ...ciphertext,
+        ...padding.subarray(0, 8),
```

Comment:
ah interesting I forgot about the padding. We could also think about replacing this with the prefix, in case we decide we don't need the prefix for asym encryption.

Reply:
yeah, I thought about this I remember we talked about this, let's agree on the best option!